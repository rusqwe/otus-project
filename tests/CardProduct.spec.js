import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import component from "src/components/CardProduct.vue";
describe('CardProduct component', () => {
  const item = {
    product: {
      "id": 4,
      "title": "",
      "price": 15.99,
      "description": "",
      "category": "",
      "image": "",
      "count": 2
    }
  };
  let wrapper;
  beforeEach(() => {
    wrapper = mount(component, { props: item });
  })
  it('mount without errors', () => {
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  })
  it('render price to list', () => {
    const price = wrapper.find('.price');
    expect(price.element.innerHTML).toEqual('31.98$')
  })
})