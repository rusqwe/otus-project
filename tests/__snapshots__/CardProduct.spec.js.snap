// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardProduct component > mount without errors 1`] = `
"<div data-v-d3eb8fbd=\\"\\" class=\\"product-item\\">
  <div data-v-d3eb8fbd=\\"\\" class=\\"product-img\\"><img data-v-d3eb8fbd=\\"\\" src=\\"\\" height=\\"115\\"></div>
  <div data-v-d3eb8fbd=\\"\\" class=\\"product-list\\">
    <h3 data-v-d3eb8fbd=\\"\\"></h3><span data-v-d3eb8fbd=\\"\\" title=\\"\\"></span>
  </div><span data-v-d3eb8fbd=\\"\\" class=\\"price\\">31.98$</span>
</div>"
`;
